plugin.js

```diff
@@ -115,20 +116,32 @@ class CspHtmlWebpackPlugin {
 
     const policyObj = JSON.parse(JSON.stringify(this.policy));
 
-    const inlineSrc = $('script:not([src])')
-      .map((i, element) => this.hash($(element).html()))
-      .get();
-    const inlineStyle = $('style:not([href])')
-      .map((i, element) => this.hash($(element).html()))
-      .get();
-
     // Wrapped in flatten([]) to handle both when policy is a string and an array
-    policyObj['script-src'] = flatten([policyObj['script-src']]).concat(
-      inlineSrc
-    );
-    policyObj['style-src'] = flatten([policyObj['style-src']]).concat(
-      inlineStyle
-    );
+    const flattenedPolicyObjScriptSrc = flatten([policyObj['script-src']]);
+    const flattenedPolicyObjStyleSrc = flatten([policyObj['style-src']]);
+
+    const { hashIfUnsafeInlinePresent } = this.opts;
+
+    if (
+      hashIfUnsafeInlinePresent ||
+      !flattenedPolicyObjScriptSrc.includes("'unsafe-inline'")
+    ) {
+      const inlineSrc = $('script:not([src])')
+        .map((i, element) => this.hash($(element).html()))
+        .get();
+      policyObj['script-src'] = flattenedPolicyObjScriptSrc.concat(inlineSrc);
```

Comment:
Let's move creating this into it's own function - it can have the interface:

```
createObjPolicy($, selector) {
  return $(selector)
    .map((i, element) => this.hash($(element).html()))
    .get();
}
```

Then we can add (pseudo):

```
policyObj['style-src'] = hasUnsafeInline ? don't do anything: this.createPolicyObj($, 'style:not([href])');
etc.
```